src/friktion/infos.ts

```diff
@@ -340,4 +353,51 @@ export class VaultInfoWrapper {
   getAPY(): number {
     return this.vaultInfo.snapshotInfo ? this.vaultInfo.snapshotInfo.apy : 0;
   }
+  async getSharePrice(
+    connection: Connection,
+    round?: number,
+    claimDeposit?: boolean,
+    claimWithdrawal?: boolean
+  ): Promise<number> {
+    if (this.vaultInfo.snapshotInfo && this.vaultInfo.roundNumber.toNumber() == round) {
+      return this.vaultInfo.snapshotInfo.shareTokenPrice;
+    } else if (this.vaultInfo.snapshotInfo) {
```

Comment:
This condition can be combined with the first condition?

Reply:
Need this because if the first condition is `||` instead of `&&`, the `snapshotInfo` might be `undefined`, still need another check.

Reply:
The first condition is redundant. Imo should use nested if/else for better readability. Otherwise the condition is confusing. For example:

```
} else if (round) {
...
```

which ignores the fact that `snapshot == undefined` has to be satisfied. 


Plz refactor to: (or similar logic if anything is mis-considered)

```typescript
if (vaultInfo.snapshotInfo) {
  // Has snapshot. No need to deal with roundNumber.
  // Return price directly
  return snapshotInfo.shareTokenPrice
} else {
  // No snapshot. Need to compute price here
  if (round) {
    // Has round. Compute price here
    if (isDeposit) {
      return price
    } else {
     return price
    }
  } else {
    // No round. Compute round here
    // Compute price directly
  }
}
```

Reply:
There's another case that have `round` but the round is current round. so it need to return the price in snapshot.

Reply:
Isn't this case covered by case 1 above?

Reply:
In your snippet there didn't cover the case that round is provided and snapshot exist.
If round is provided and it's not current round, then it goes into isDeposit case.

Reply:
If snapshot exists, is there any difference in price btw the following 2 cases:

- round is provided
- round is not provided

In your original implementation both cases return `snapshotInfo.shareTokenPrice`

Reply:
Yes there's a different when round is provided it also need that the round is current round to return `snapshotInfo.shareTokenPrice`

Reply:
I see, my first version ignore the case that round is provided before checking `snapshotInfo` exist.

Reply:
This version is correct now